### ISSUE TITLE
Extending default actions

### ISSUE THREAD

**Issue by colepanike** [balderdashy/sails#2343]
I would love to have the ability to extend the base actions like `find`,`create`, etc. for a specific module without having to override the whole blueprint. I understand this used to be possible via `next()` (according to [this answer](http://stackoverflow.com/a/18113698/1795429) on StackOverflow by Mike). This is not functional in v.10, at least as far as I am able to determine.

I do not know the term for the object to which I will refer now but I will call it the `inherited base` from here on out. The object I'm talking about is the one that must be called whenever I call something like `User.find()` when I haven't overridden it.

What I propose is that the `inherited base` possess a `base` object that holds all of these default actions. The `inherited base` should then make the top level action (the one that is currently called by `find`, `create`, etc.) into an alias to the `base` method. Thus only the alias has to be fully overridden.

Under this approach a developer could write a new `find` action and incorporate the `base` method `User.base.find()` and use standard waterline queries like he/she could in any other action. This enables things like deep population for each call to the base find and increases the usefulness of the default action routes.

I've not had the time to really dig into the source so I'm speaking from a somewhat uninformed perspective. Is this even possible without a major restructure? I'm thinking that if it was possible before via `next()` then it shouldn't be too hard this way.

**Comment by CWyrtzen** [comment]
@colepanike was this resolved with a new release? it could also be marked as a FR~

**Comment by sailsbot** [comment]
Thanks for posting, @colepanike.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
